tests/test_hdcache.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2021-2023 Lawrence Livermore National Security, LLC and other
+ * AMSLib Project Developers
+ *
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <AMS.h>
+
+#include <cstring>
+#include <iostream>
+#include <ml/hdcache.hpp>
+#include <umpire/ResourceManager.hpp>
+#include <umpire/Umpire.hpp>
+#include <vector>
+#include <wf/resource_manager.hpp>
+
+template <typename T>
+std::vector<const T *> generate_vectors(const int num_clusters,
+                                        int elements,
+                                        int dims)
+{
+  std::vector<const T *> v_data;
+  // This are fixed to mimic the way the faiss was generated
+  // The code below generates data values that are either within
+  // the distance of the faiss index or just outside of it.
+  const T distance = 10.0;
+  const T offset = 5.0;
+  for (int i = 0; i < dims; i++) {
+    T *data = ams::ResourceManager::allocate<T>(num_clusters * elements,
+                                                AMSResourceType::HOST);
+    for (int j = 0; j < elements; j++) {
+      // Generate a value for every cluster center
+      for (int k = 0; k < num_clusters; k++) {
+        T tmp = ((T)rand()) / INT_MAX;
+        tmp += (k + 1) * num_clusters;
+        if (j % 2 == 0) {
+          tmp += offset;
+        }
+        data[j * num_clusters + k] = tmp;
+      }
+    }
+    v_data.push_back(data);
+  }
+  return std::move(v_data);
+}
+
+template <typename T>
+void print_vectors(std::vector<T *> &vec, int num_elements, int num_clusters)
+{
+  for (int i = 0; i < num_elements; i++) {
+    for (int c = 0; c < num_clusters; c++) {
+      for (auto v : vec) {
+        std::cout << v[i * num_clusters + c] << ":";
+      }
+      std::cout << std::endl;
```

Comment:
Is there a particular reason to flush the output here with `std::endl`?

Reply:
Fixed